src/System.Private.CoreLib/shared/System/IO/BinaryReader.cs

```diff
@@ -649,6 +578,42 @@ public virtual byte[] ReadBytes(int count)
             return result;
         }
 
+        private ReadOnlySpan<byte> InternalRead(int numBytes)
+        {
+            Debug.Assert(numBytes >= 2 && numBytes <= 16, "value of 1 should use ReadByte. value > 16 requires to change the minimal _buffer size");
+
+            if (_isMemoryStream)
+            {
+                // no need to check if _stream == null as we will never have null _stream when _isMemoryStream = true
+
+                // read directly from MemoryStream buffer
+                Debug.Assert(_stream is MemoryStream);
+                return ((MemoryStream)_stream).InternalReadSpan(numBytes);
+            }
+            else
+            {
+                if (_stream == null)
+                {
+                    throw Error.GetFileNotOpen();
+                }
+
+                int bytesRead = 0;
+                int n = 0;
+
+                do
+                {
+                    n = _stream.Read(_buffer, bytesRead, numBytes - bytesRead);
+                    if (n == 0)
+                    {
+                        throw Error.GetEndOfFile();
+                    }
+                    bytesRead += n;
+                } while (bytesRead < numBytes);
+
+                return _buffer;
+            }
+        }
+
         protected virtual void FillBuffer(int numBytes)
```

Comment:
It may be useful to add a comment on top of FillBuffer that explains the history on why it is no longer called.

Reply:
fixed.